lib/manager/kustomize/readme.md

```diff
@@ -14,16 +14,51 @@ This package will manage two parts of the `kustomization.yaml` file:
 
 - Needs to have `kind: Kustomization` defined
 - Currently this hasn't been tested using HTTPS to fetch the repos
-- The image tags are limited to the following formats:
+- Order of image tags does not matter:
```

Comment:
```suggestion
- The keys for the image tags can be in any order
```

Is this more correct? I'm trying to wrap my head around all the discussion that happened in the PR... If it's bad don't apply the suggestion!

If it is _required_ to have a `name:` key we should document that maybe? Or maybe it's really obvious you need a `name:` key or `kustomize` won't work at all?

---

Also can you check that the comments/ideas here are properly reflected in the docs? I think not all of the points you're making in this comment are in the docs?

> > @thepwagner did you saw my previous work here: #8110 ?
> 
> Yep! Sorry, I should have mentioned that. I liked the pattern of a [regex to match multiple YAML keys](https://github.com/renovatebot/renovate/pull/8110/files#diff-967765bec1a1a1349e489e39fb7889610ad5381ba2ba6c13454f25ceed6037b8R38). I figured that pattern would need to include `newName` as well, and didn't like the prospect of `abc|acb|bac|bca|cab|cba` (since YAML will allow the values in any order).
> 
> That's what sent me look at [pre-splitting the `images:` block as a string](https://github.com/thepwagner/renovate/pull/2/files#diff-967765bec1a1a1349e489e39fb7889610ad5381ba2ba6c13454f25ceed6037b8R116-R117). Key order wouldn't matter, as long as there was a `name:` in there somewhere the `replaceString` contain the fully indented `.images[]` entry.
> 
> I abandoned the latter approach as [`autoReplaceStringTemplate`](https://github.com/thepwagner/renovate/pull/2/files#diff-967765bec1a1a1349e489e39fb7889610ad5381ba2ba6c13454f25ceed6037b8R64) felt necessary to support injecting `newDigest` if it didn't already exist (which would afford `pinDigests: true` and fully close #8089). Doing this while maintaining the user's original key order did not seem like fun. I may have been to focused on maintaining key order. While it reads as weird to me, the following _should_ be supported, right?
> 
> ```yaml
>   - newTag: 3.12.2
>     name: alpine
>     digest: sha256:25f5332d060da2c7ea2c8a85d2eac623bd0b5f97d508b165f846c7d172897438
>   - digest: sha256:e1488cb900233d035575f0a7787448cb1fa93bed0ccc0d4efc1963d7d72a8f17
>     newTag: 1.32.1
>     name: busybox:1.30.0
>   - newName: amd64/busybox:1.30.1
>     name: busybox
>     digest: sha256:e1488cb900233d035575f0a7787448cb1fa93bed0ccc0d4efc1963d7d72a8f17
> ```
> 
> So when I realized `newTag: tag@digest` worked for my use case, I gave up on multiple keys and started on this priority-driven approach: `digest > newTag > newName`. It seems correct, but admittedly punishes users that provide too much information that kustomize will ignore. Per the discussion with _rarkins, I think nudging those users towards `newTag: tag@digest` is a better course.